packages/nuxt/src/core/nitro.ts

```diff
@@ -594,3 +593,34 @@ async function spaLoadingTemplate (nuxt: Nuxt) {
 
   return ''
 }
+
+async function resolvePublicFolders (nuxt: Nuxt) {
+  const { dir: rootDir, srcDir, alias: rootAlias } = nuxt.options
+  const dirs = await Promise.all(nuxt.options._layers.map((layer) => {
+    const [path, cwd, alias] = layer.config.rootDir === nuxt.options.rootDir
+      ? [rootDir.public || 'public', srcDir, rootAlias]
+      : [layer.config.dir?.public || 'public', layer.config.srcDir, layer.config.alias]
+    return resolvePath(path, { cwd, alias: alias as Record<string, string> })
+  }))
+
+  return dirs.filter(dir => existsSync(dir)).map(dir => ({ dir }))
+}
+
+async function resolvePublicAssets (nuxt: Nuxt) {
+  const { dir: rootDir, srcDir, alias: rootAlias } = nuxt.options
+  await Promise.all(nuxt.options._layers.map((layer) => {
+    return Promise.all((layer.config.nitro?.publicAssets || []).map(async (publicAsset) => {
+      if (!publicAsset) {
```

Comment:
shold we use `!publicAssets?.dir` instead: if so we should remove `publicAsset.dir || ` from L617 and L618

Reply:
I've exluded `publicAssets` without `dir`